pkg/query/storeset.go

```diff
@@ -0,0 +1,204 @@
+package query
+
+import (
+	"context"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/improbable-eng/thanos/pkg/cluster"
```

Comment:
Did the query pkg depend on the cluster pkg before?

Reply:
No ): Can mitigate that by introducing some interface here.